### PR TITLE
Adds WORKFLOW.md file

### PR DESCRIPTION
**Pull request by urimandujano**
I'm basing this document heavily off of the document that Kurt has been writing up in our service repositories. The biggest differences are that I've added references (where appropriate) to ReadTheDocs, removed the release script, made references to when and where to create PRs during the release process, and filled in the `Deploying to production` section with a reference to a currently non-existent Github Action.

The end goal is to have a workflow here that is very similar to our service repos. This process is what I had in mind when making my comments on PR-251 in Flows. I'm hoping we can compare and make updates to both PRs so that we're happy with the workflow.

### CODE REVIEW

**Review by jakeglobus — APPROVED** [review]

**Comment by kurtmckee** [comment]
@urimandujano I'm waiting to review this until after I've merged the work elsewhere on this doc. In particular, I'm working on making a GitHub action that can perform checks when attempting to merge to production.

**Comment by jimPruyne** [comment]
Shall we get the globusonline/flows#251 settled first and then take another pass on this one? Or, is the stuff pulled from there good enough and we should review separately at this point?

**Comment by urimandujano** [comment]
@jimPruyne If we get the Flows PR settled then I'd like for us to hop back over here and make sure the process in this doc is consistent with what we have there. And if not then at least understanding that it'll be inconsistent (:

File: WORKFLOW.md
Comment (kurtmckee) [inline WORKFLOW.md:41]: It may be useful to copy in the text regarding Shortcut integration from the flows repo's workflow doc.

File: WORKFLOW.md
Comment (kurtmckee) [inline WORKFLOW.md:56]: > Feature releases begin by creating a new branch off of `main`

The text in the flows repo now points out that in some cases it's necessary to target a specific SHA, rather than `origin/main` directly. This might not be necessary here but wanted to point it out.

File: WORKFLOW.md
Comment (kurtmckee) [inline WORKFLOW.md:61]: `"release-..." --> "release/..."`

File: WORKFLOW.md
Comment (kurtmckee) [inline WORKFLOW.md:80]: `"hotfix-..." --> "hotfix/..."`

File: WORKFLOW.md
Comment (kurtmckee) [inline WORKFLOW.md:118]: I think should be expanded, similar to the doc that landed in the flows repo.

It may be appropriate to add some of the "update the project metadata" commands so those can be referenced when prepping a release.

File: WORKFLOW.md
Comment (kurtmckee) [inline WORKFLOW.md:125]: Are these already in place? When I released a new version I needed to run

```
poetry build
poetry publish
```

If it's not automated then this should be expanded, possibly including the git commands to pull and checkout the `production` branch. Something like this:

```shell
git checkout production
git pull --tags
poetry build
poetry publish
```

**Review by kurtmckee — CHANGES_REQUESTED** [review]

File: WORKFLOW.md
Comment (urimandujano) [reply WORKFLOW.md]: Good point, I added steps for how to build and publish to PyPI. I also removed all the text about where a GH Action would take over during the release. I figure whenever we create that GH Action we can document it here and remove redundant steps.

File: WORKFLOW.md
Comment (urimandujano) [reply WORKFLOW.md]: This one was a little more freehand but I think I got into a decent level of detail in this section in the doc including some commands that should be run along the way. Can you take a look and let me know if that's what you were thinking?

File: WORKFLOW.md
Comment (urimandujano) [reply WORKFLOW.md]: I added a comment to this point and a script snippet for how to create a branch off a SHA.

File: WORKFLOW.md
Comment (urimandujano) [reply WORKFLOW.md]: Done!

**Comment by urimandujano** [comment]
@jimPruyne @kurtmckee @rudyardrichter I just came back here to update based on edits that were made into `WORKFLOW.md` in the Flows repo. A few of us have gone through this process already so please let me know if you read something and it looks unfamiliar. I'd like to get this doc settled and copied into action-provider-tools by the end of this sprint.

File: WORKFLOW.md
Comment (jimPruyne) [inline WORKFLOW.md:45]: I'm actually a bit on the fence here. As this is an open source repo, I 1) Don't think automation is setup and 2) I'm not sure we want to expose the Shortcut numbers etc. as that provides no useful info to anyone outside the org looking at this repo.

**Review by jimPruyne — COMMENTED** [review]
Text looks good. I guess we should discuss how/if we want to do SC integration on this repo.

File: WORKFLOW.md
Comment (kurtmckee) [reply WORKFLOW.md]: I think including the story number in the branch name doesn't cost much but I'm fine removing this additional information here.

I do think we should automate the Shortcut integration, though.

File: WORKFLOW.md
Comment (kurtmckee) [inline WORKFLOW.md:118]: For semantic versioning I think this should be something like:

```markdown
        - If the release is a hotfix, use 
          `poetry version patch`
        - If the release is a backwards-compatible change, use 
          `poetry version patch`
        - If the release is not backwards compatible, use 
          `poetry version major`
```

File: WORKFLOW.md
Comment (kurtmckee) [inline WORKFLOW.md:157]: I think GH Markdown supports a code block even within this context:

## Code

```markdown
   2. Code updates need to be published to PyPI. To do so, the following steps
      will need to be performed on a Terminal window:

      ```shell
      git checkout production
      git pull --tags
      poetry publish --build
      ```
```

## Result

2. Code updates need to be published to PyPI. To do so, the following steps
   will need to be performed on a Terminal window:

   ```shell
   git checkout production
   git pull --tags
   poetry publish --build
   ```

File: WORKFLOW.md
Comment (kurtmckee) [inline WORKFLOW.md:150]:
```diff
- updates will be pulled direction
+ updates will be pulled directly
```

**Review by kurtmckee — CHANGES_REQUESTED** [review]

File: WORKFLOW.md
Comment (kurtmckee) [reply WORKFLOW.md]: Added some notes. Resolving this thread. Thanks for all of your work on this, Uriel!

File: WORKFLOW.md
Comment (urimandujano) [reply WORKFLOW.md:118]: The SDK/CLI is _technically_ still in beta (it's a 0.X.X version) so while bumping the major version on non-backward's compatible changes is right, I don't think we can do that yet. I will add a case handling how to deal with hotfix versioning.

Comment (kurtmckee) [reply WORKFLOW.md:118]: Oh, yep, that's right. Okay, skip that bit about the major version. =)

File: WORKFLOW.md
Comment (urimandujano) [reply WORKFLOW.md:157]: Let me try that out and see how it gets rendered ....

Comment (urimandujano) [reply WORKFLOW.md]: You are 100% right about that! I feel like I've been learning a lot of markdown in reviews lately so thank you for that!

**Review by kurtmckee — APPROVED** [review]

**Pull request merged**
